app/scripts/modules/core/src/pipeline/config/stages/googleCloudBuild/GoogleCloudBuildStageForm.tsx

```diff
@@ -127,94 +134,114 @@ export function GoogleCloudBuildStageForm(props: IGoogleCloudBuildStageFormProps
     }
   }, [stage.triggerType]);
 
+  const freeformSpelEnabled = React.useMemo(
```

Comment:
What other configuration are you anticipating in `FormikSpelContext`?  Seeing this boilerplate and the usage down below (`spelConfig={freeformSpelDisabled}`) makes me wish for a simple `spelAware={false}`.

If the expectation is that we will add further things to the Spel context (i.e.: `{ freeformSpelDisabled: true, otherFieldToggle: false, thirdtoggle: true }`) then I have followup questions:  

- Would each component that provides this context object have opinions for _all the spel settings_ the context object contains?  For example, if we have three toggles but I only have an opinion about freeformSpelDisabled then I have to fetch the current context and merge in my settings.  
- How would you feel about an API where each setting was a separate context with a separate override prop?
- If we like the configuration-object approach, could these be pulled up into a `SpinFormikConfigContext` instead?

Reply:
Thanks so much for the thorough review!

I was anticipating future configuration options for the following:
- Enabling a `VariablePicker` mode
- Enabling messages like the ones in Greg's mocks encouraging users to migrate SpEL to the Evaluate Variables stage

But I think you're right that each of these can just be a separate context with a separate override prop, and we can re-evaluate merging into a single config object if it becomes too many contexts/props.

Reply:
Updated so that the `FormikSpelContext` and `FormikFormField`'s `spelAware` prop are a bool rather than a config object.